src/components/enterprise-redirects/EnterpriseLearnerFirstVisitRedirect.jsx

```diff
@@ -1,24 +1,41 @@
-import React, { useEffect } from 'react';
+import React, { useContext, useEffect } from 'react';
 import { Redirect, useParams } from 'react-router-dom';
 import Cookies from 'universal-cookie';
+import { ASSIGNMENT_TYPES } from '../dashboard/main-content/course-enrollments/CourseEnrollments';
+import { UserSubsidyContext } from '../enterprise-user-subsidy';
+
+export const isFirstDashboardPageVisit = () => {
+  const cookies = new Cookies();
+
+  const hasUserVisitedDashboard = cookies.get('has-user-visited-learner-dashboard');
+  return !hasUserVisitedDashboard;
+};
 
 const EnterpriseLearnerFirstVisitRedirect = () => {
   const { enterpriseSlug } = useParams();
+  const {
+    redeemableLearnerCreditPolicies,
+  } = useContext(UserSubsidyContext);
 
-  const cookies = new Cookies();
-
-  const isFirstVisit = () => {
-    const hasUserVisitedDashboard = cookies.get('has-user-visited-learner-dashboard');
-    return !hasUserVisitedDashboard;
+  // TODO: Refactor to DRY up code for redeemableLearnerCreditPolicies
+  const hasActiveCourseAssignments = (learnerCreditPolicies) => {
+    const learnerContentAssignmentsArray = learnerCreditPolicies?.flatMap(
+      item => item?.learnerContentAssignments || [],
+    );
+    // filters out course assignments that are not considered active
+    const hasActiveAssignments = learnerContentAssignmentsArray.filter(
+      assignment => assignment.state !== ASSIGNMENT_TYPES.cancelled,
```

Comment:
[sanity check] I believe this logic currently assumes that an `errored` assignment type ([source](https://github.com/openedx/enterprise-access/blob/c1cec6a2ce2fa903384f28caff6b96d8d650760a/enterprise_access/apps/content_assignments/constants.py#L6-L13)) is valid/active? Is this expected?

Cross-checking with the logic of what assignment states the dashboard UI handles, it looks like this logic should support only `allocated` and `cancelled` states, as these are the only ones where assignments-related UI may appear in the dashboard route.

Reply:
Created a utils function as suggested to abstract the logic to a single source of truth along with referencing a single constant value for `ASSIGNMENT_TYPES`.